src/backend/core/api/viewsets/maildomain.py

```diff
@@ -202,12 +212,40 @@ def create(self, request, *args, **kwargs):
                 role=models.MailboxRoleChoices.ADMIN,
             )
 
+            contact, _ = models.Contact.objects.get_or_create(
+                email=email,
+                mailbox=mailbox,
+                defaults={"name": f"{first_name} {last_name}"},
+            )
+            mailbox.contact = contact
+            mailbox.save()
+
+        elif mailbox_type == "shared":
+            email = f"{local_part}@{domain.name}"
+            name = metadata.get("name")
+            contact, _ = models.Contact.objects.get_or_create(
+                email=email,
+                mailbox=mailbox,
+                defaults={"name": name},
+            )
+            mailbox.contact = contact
+            mailbox.save()
+
         serializer = self.get_serializer(mailbox)
         headers = self.get_success_headers(serializer.data)
         payload = serializer.data
-        if mailbox_type == "personal" and settings.IDENTITY_PROVIDER == "keycloak":
+
+        # This is a somewhat hacky bypass of abstractions, but for now
+        # we need to return a one time password synchronously
+        if (
+            mailbox_type == "personal"
+            and settings.IDENTITY_PROVIDER == "keycloak"
+            and is_identity
+            and domain.identity_sync
+        ):
```

Comment:
Checking `mailbox_type` value and `is_identity` is redundant

```suggestion
        if (
            is_identity
            and settings.IDENTITY_PROVIDER == "keycloak"
            and domain.identity_sync
        ):
```